php-checks/src/main/java/org/sonar/php/checks/security/XxeCheck.java

```diff
@@ -80,8 +92,31 @@ private void checkSetParserProperty(FunctionCallTree call) {
     }
   }
 
-  private void createIssue(Tree tree) {
-    context().newIssue(this, tree, "Disable access to external entities in XML parsing.");
+  private void checkXmlBuildOption(ExpressionTree valueTree, Tree primaryTree) {
+    if (valueTree.is(Kind.VARIABLE_IDENTIFIER)) {
+      CheckUtils.uniqueAssignedValue((VariableIdentifierTree) valueTree).ifPresent(x -> checkXmlBuildOption(x, primaryTree));
```

Comment:
Due to this recursion, we can fall into a dead loop which will end up in a StackOverflowError.
```
  $options3 = $options4;
  $options4 = $options3;

  $xml = Xml::build($content, $options3);
```
Of course, this is not executable PHP code. But we parse it and we should prevent to crash on analysis, even when the code is not valid. I would recommend checking if the result of uniqueAssignedValue is not the original expression.